doc/source/guides/common-tasks.rst

```diff
@@ -0,0 +1,116 @@
+.. _guide-common-tasks:
+
+==================
+Working with Tasks
+==================
+
+The Gini iOS SDK makes heavy use of the concept of tasks. Tasks are convenient when you want to
+do a series of tasks in a row, each one waiting waiting for the previous to finish (comparable to
+Promises in JavaScript). This is a common pattern when working with Gini's remote API.
+The Gini iOS SDK uses `facebook's task implementation, which is called bolts <https://github.com/BoltsFramework/Bolts-iOS>`_.
+Before you continue reading this guide, we strongly encourage you to read the `short guide for the Bolts
+framework <https://github.com/BoltsFramework/Bolts-iOS/blob/master/README.md#tasks>`_.
+
+Upload a document
+=================
+
+As the key aspect of the Gini API is to provide information extraction for analyzing documents, the
+API is mainly built around the concept of documents. A document can be any written representation
+of information, usually such as invoices, reminders, contracts and so on.
+
+The Gini iOS SDK supports creating documents from images, usually a picture of a paper document
+which was taken with the device's camera. The following example shows how to create a new
+document from an image.
+
+
+.. code-block:: obj-c
+
+    #import <Gini-iOS-SDK/GiniSDK.h>
+
+    ...
+
+    // Assuming that `gini` is an instance of the `GiniSDK` facade class and `image` is an `UIImage` instance,
```

Comment:
a `UIImage` instance (since the U is pronounced 'ju:', we don't need 'an' but 'a')